app/src/ui/merge-conflicts/unmerged-file.tsx

```diff
@@ -0,0 +1,294 @@
+import * as React from 'react'
+import {
+  isConflictWithMarkers,
+  isManualConflict,
+  ConflictedFileStatus,
+  ConflictsWithMarkers,
+} from '../../models/status'
+import { join } from 'path'
+import { Repository } from '../../models/repository'
+import { Dispatcher } from '../dispatcher'
+import { showContextualMenu } from '../main-process-proxy'
+import { Octicon, OcticonSymbol } from '../octicons'
+import { PathText } from '../lib/path-text'
+import {
+  ManualConflictResolutionKind,
+  ManualConflictResolution,
+} from '../../models/manual-conflict-resolution'
+import {
+  OpenWithDefaultProgramLabel,
+  RevealInFileManagerLabel,
+} from '../lib/context-menu'
+import { openFile } from '../lib/open-file'
+import { shell } from 'electron'
+import { Button } from '../lib/button'
+import { IMenuItem } from '../../lib/menu-item'
+import { LinkButton } from '../lib/link-button'
+
+export const renderUnmergedFile: React.SFC<{
+  readonly repository: Repository
+  readonly path: string
+  readonly status: ConflictedFileStatus
+  readonly manualResolution?: ManualConflictResolution
+  readonly resolvedExternalEditor: string | null
+  readonly openFileInExternalEditor: (path: string) => void
+  readonly dispatcher: Dispatcher
+}> = props => {
+  if (isConflictWithMarkers(props.status)) {
+    return renderConflictedFileWithConflictMarkers({
+      path: props.path,
+      status: props.status,
+      resolvedExternalEditor: props.resolvedExternalEditor,
+      onOpenEditorClick: () =>
+        props.openFileInExternalEditor(join(props.repository.path, props.path)),
+      repository: props.repository,
+      dispatcher: props.dispatcher,
+    })
+  }
+  if (isManualConflict(props.status) && props.manualResolution === undefined) {
+    return renderManualConflictedFile({
+      path: props.path,
+      repository: props.repository,
+      dispatcher: props.dispatcher,
+    })
+  }
+  return renderResolvedFile({
+    path: props.path,
+    repository: props.repository,
+    dispatcher: props.dispatcher,
+    manualResolution: props.manualResolution,
+  })
+}
+
+const renderResolvedFile: React.SFC<{
+  readonly repository: Repository
+  readonly path: string
+  readonly manualResolution?: ManualConflictResolution
+  readonly dispatcher: Dispatcher
+}> = props => {
+  let statusString = 'No conflicts remaining'
```

Comment:
:art: this could be it's own function to avoid the `let` and inline checks:

```ts
function getStatusString(manualResolution: ManualConflictResolutionKind) {
  if (props.manualResolution === ManualConflictResolutionKind.ours) {
    return 'Using our version'
  }
  if (props.manualResolution === ManualConflictResolutionKind.theirs) {
    return 'Using their version'
  }
  return 'No conflicts remaining'
}

Reply:
super into that, was thinking the same when i wrote that :)

Reply:
agree inline checks like this are 😢